R/provider-azure.R

```diff
@@ -58,7 +58,7 @@ ProviderAzure <- new_class(
   parent = ProviderOpenAI,
   properties = list(
     api_key = prop_string(),
-    token = prop_string(allow_null = TRUE),
+    token = prop_azure_token(),
```

Comment:
Since we're only using this in a single place, there's no need to pull out into a separate function. And indeed, since we expect that pretty much all `ProviderAzure` objects are created by `chat_azure()`, I'd just do any validation there and not worry about it in the class.

Reply:
Agreed. That is a good idea. I added a local version of `is_azure_token` from the AzureAuth package to utils, and removed the class validation from utils-S7. Now the validation is done in the `chat_azure` function and is validated as a string in the ProviderAzure class. 

I also renamed the argument token to azure_token, because I thought it was confusing with the term "token" being used for LLM tokens.  Once the access_token is extracted from the object, I called it access_token to make it clear that it was no longer the azure_token object.